File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        stringstream ss(s);
        vector<string>temp;
        string word;
        while(ss>>word){
            temp.push_back(word);
        }
        reverse(temp.begin(),temp.end());
        string ans="";
        for(int i=0;i<temp.size();i++){
            for(int j=0;j<temp[i].size();j++){
                ans+=temp[i][j];
            }
            ans+=" ";
        }
        ans.pop_back();
        return ans;
        
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        int n=s.length();

        s = new StringBuilder(s).reverse().toString();
        String ans="";
        for(int i=0;i<n;i++){
            String word="";

            while(i<n && s.charAt(i)!=' '){
                word+=s.charAt(i);
                i++;
            }
            word=new StringBuilder(word).reverse().toString();
            if(word.length()>0){
                ans+=" "+word;
            }
        }

        return ans.substring(1);
    }
}